### PR TITLE
Reduce fee queries

### PR DESCRIPTION
**Pull request by krzysztof-jelski**
# Checklist:

- [x] Code follows the style guidelines
- [x] Documentation updated
- [x] Corner cases evaluated and handled
- [x] Automatic tests added/update
- [x] Automatic tests pass
- [x] Software runs (locally or hosted)
- [x] Sanity check passed
- [x] Self-review passed

### CODE REVIEW

File: src/builder.js [hunk 1]
```diff
@@ -42,6 +42,7 @@ import ChallengesRepository from './services/challenges_repository';
 import Migrator from './migrations/Migrator';
 import FailedChallengesCache from './services/failed_challenges_cache';
 import WorkerTaskTrackingRepository from './services/worker_task_tracking_repository';
+import * as Sentry from '@sentry/node';
```
Comment (vanruch) [inline src/builder.js]: Why `import * as Sentry` when we always use just `import Sentry from ...`?

File: src/services/upload_repository.js [hunk 2]
```diff
@@ -10,28 +10,26 @@ This Source Code Form is “Incompatible With Secondary Licenses”, as defined
...
   }
 
   async ensureBundleIsUploaded(bundleId, storagePeriods) {
     await this.validateNodeIsOnboardedAsHermes();
-    await this.validateEnoughFundsForUploadFee(storagePeriods);
 
     const previousUploader = await this.shelteringWrapper.getBundleUploader(bundleId);
     if (this.isEmptyAddress(previousUploader)) {
-      return this.receiptWithResult(this.uploadsActions.uploadBundle(bundleId, storagePeriods), 'Bundle has been uploaded');
+      return this.receiptWithResult(this.uploadsActions.uploadBundle(bundleId, storagePeriods, this.lowFundsWarningAmount), 'Bundle has been uploaded');
```
Comment (vanruch) [inline src/services/upload_repository.js]: `lowFundsWarningAmount` should be passed to the UploadActions' constructor, this shall not work now

File: src/services/upload_repository.js [hunk 3]
```diff
@@ -40,27 +38,35 @@ export default class UploadRepository {
...
+      this.handleLowBalanceWarning(uploadReceipt);
+      return {
+        ...uploadReceipt,
+        uploadResult
+      };
+    } catch (error) {
+      if (error instanceof InsufficientFundsToUploadBundleError) {
+        this.sentry.captureException(error);
+        throw error;
+      }
```
Comment (vanruch) [inline src/services/upload_repository.js]: I'd move rethrow outside of if, cause if we will catch some other type of error, it will be ignored

File: test/services/upload_repository.js [hunk 4]
```diff
@@ -60,22 +58,34 @@ describe('Upload repository', () => {
...
     });
 
     it('throws if not enough funds', async () => {
-      web3Mock.eth.getBalance.resolves(tooSmallBalance);
+      uploadsActionsMock.uploadBundle = sinon.stub().rejects(new InsufficientFundsToUploadBundleError(fee, tooSmallBalance));
       await expect(uploadRepository.ensureBundleIsUploaded(bundleId, storagePeriods)).to.be.eventually.rejected;
+      expect(sentryMock.captureException).to.be.calledOnce;
+    });
+
+    it('sentry log if low balance', async () => {
```
Comment (vanruch) [inline test/services/upload_repository.js]:
```suggestion
    it('logs to sentry low balance warning', async () => {
```
